### ISSUE TITLE
cms.utils.permissions - get_user_permission_level - AttributeError 'Page' has no attribute 'level'

### ISSUE THREAD

**Issue by benjhastings** [divio/django-cms#4928]
As the title suggests, the "get_user_permission_level" function in cms.utils.permissions tries to return the level of the page and falls over with an AttributeError.  I don't think this is something that I have messed up (I am adding the CMS to an existing project that was on django 1.5, upgrading to django 1.8.8 in the process so I have to contend with the new migrations).
This is using version 3.2.0 and occurs on line 202.
I *think* that is is just a case of changing .level to .get_depth() but I don't know what knock on effects this will have.

Furthermore there are two lookups for __level in get_subordinate_users() and get_subordinate_groups() in the same file. (Let me know if you need new tickets raising for these)

```python
def get_user_permission_level(user):
    """
    Returns highest user level from the page/permission hierarchy on which
    user haves can_change_permission. Also takes look into user groups. Higher
    level equals to lover number. Users on top of hierarchy have level 0. Level
    is the same like page.level attribute.

    Example:
                              A,W                    level 0
                            /    \
                          user    B,GroupE           level 1
                        /     \
                      C,X     D,Y,W                  level 2

        Users A, W have user level 0. GroupE and all his users have user level 1
        If user D is a member of GroupE, his user level will be 1, otherwise is
        2.

    """
    if (user.is_superuser or
            GlobalPagePermission.objects.with_can_change_permissions(user).exists()):
        # those
        return 0
    try:
        permission = PagePermission.objects.with_can_change_permissions(user).order_by('page__path')[0]
    except IndexError:
        # user isn't assigned to any node
        raise NoPermissionsException
    return permission.page.level #<----OFFENDING LINE

**Comment by jbazik** [comment]
I just bumped into this.   Treebeard MP_Node has a "depth" attribute, so it looks like the fix is to simply replace level with depth.  What I can't understand is how this has escaped fixing all this time, and I see new code in develop that still references "level."  Am I missing something?

The level attribute is also referenced in some templates, possibly elsewhere.  I can provide a PR if someone can confirm that this hasn't been fixed some other way.

**Comment by FinalAngel** [comment]
@divio/django-cms-core any feedback?

**Comment by czpython** [comment]
Confirmed. Will need to be backported to 3.1

**Comment by czpython** [comment]
I'm currently working on a fix for this on 3.3.x

**Comment by czpython** [comment]
Fixed by https://github.com/divio/django-cms/pull/5532
